source/SpinalHDL/Data types/Int.rst

```diff
@@ -150,15 +150,22 @@ Logic
 
 .. note::
 
-   ``x rotateLeft y`` and ``x rotateRight y`` are also valid syntax.
+   ``x rotateLeft y`` and ``x rotateRight y`` are also valid syntax.  Due to
+   the Scala languge able to infer a method with argument invocation from
+   this syntax.
 
 .. note::
 
    Notice the difference in behaviour between ``x >> 2``:T(w(x)-2) and ``x >> U(2)``:T(w(x))
    due to the Scala type of ``y``.
 
    The difference is that in the first case 2 is an ``Int`` (which can be seen as an
-   "elaboration integer"), and in the second case it is a hardware signal (type ``UInt``).
+   "elaboration integer constant"), and in the second case it is a hardware signal
+   (type ``UInt``) that may or may-not be a constant.  In this example it is a
+   constant described as ``U(2)`` but if the value was provided from a
+   ``Reg(UInt(2 bits))`` or other similar dynamic signal that would elborate
+   differently and has appropiate width constraints upon `y` in sympathy with the
```

Comment:
There are no constraints placed on y, see:
```scala
object Blub extends App {
  SpinalVerilog(new Component {
    val ui = in port UInt(5 bit)
    val shift = in port UInt(20 bit)
    val uol = ui >> U(2)
    val uos = ui >> shift

    uol.asOutput()
    uos.asOutput()
  })
}
```
Which builds perfectly fine (even though shift is 20 bit).

Also: I would argue that both cases elaborate the same, a signal is being used for shifting, there will be a difference after synthesis.

Nits:
- In this example it is a constant described as ``U(2)`` -> simplify to `While U(2) is a constant, if...`
- `y` and `x` have wrong code tags?
- `in sympathy with the target x being modified` -> `depending on x`?

Reply:
Just to offer the example where I did width constraint placed upon the part `y` by SpinalHDL for the rotates:

```
SpinalVerilog(new Component {
    val ui = in port Bits(4 bit)
    val shift = in port UInt(4 bit)  // This needs to be log2Up(ui.getWidth) or less
    val someBits = Bits()
    someBits := ui.rorateLeft(shift)
    // someBits := ui << shift
  })
```

Will re-read and work out what needs clarity shortly.

Reply:
Moved width constraint being explained to more specific location near respective API

Reply:
So the only remaining point is that both versions still elaborate the same

Reply:
This also related to the rotate methods, but the change in elaboration isn't anything that optimization from pruning wouldn't achieve so I removed that also.